pkgs/http/test/html/cache_test.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@TestOn('browser')
+library;
+
+import 'dart:async';
+
+import 'package:http/browser_client.dart';
+import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
+import 'package:test/test.dart';
+
+import 'utils.dart';
+
+void main() {
+  late Uri url;
+  setUp(() async {
+    final channel = spawnHybridUri(Uri(path: '/test/stub_server.dart'));
+    var port = (await channel.stream.first as num).toInt();
+    url = echoUrl.replace(port: port);
+  });
+
+  test('#send a GET with default type', () async {
+    var client = BrowserClient(cacheMode: CacheMode.defaultType);
+    await client.get(url);
+    var response = await client.get(url);
+    client.close();
+
+    expect(response.statusCode, 200);
+    expect(response.reasonPhrase, 'OK');
+    expect(response.body, parse(allOf(containsPair('numOfRequests', 2))));
```

Comment:
Shouldn't this be one? For the fetch api, do you need to manually set etags?

Reply:
Nope, 2 because it is called twice.

Check cronet cache tests as well for reference.

Reply:
What do you mean by etags ?
You mean the timeout skipping comments ?

If yes, I guess yes but overall it works.

Reply:
I thought that `CacheMode.defaultType` enabled caching? So why is the server accessed twice is caching is enabled? 

When caching is enabled in `cronet_http`, only one request is seen by the server:
https://github.com/dart-lang/http/blob/900da9fe7d576caa4882053cf26d3a70bd3106e0/pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart#L48

You can see that the test server for cronet_http (https://github.com/dart-lang/http/blob/900da9fe7d576caa4882053cf26d3a70bd3106e0/pkgs/cronet_http/example/integration_test/cronet_configuration_test.dart#L24)

sets both the `Cache-Control` and `etag` headers.

The whatwg spec has information on how etags are used:

![image](https://github.com/user-attachments/assets/db0bcb83-76fe-42b8-9610-3af5c104d5fc)

Reply:
So, I think I will push another test with different etag values.

You are correct.

Reply:
Lets call it regression test to check a time when server enables caching on a specific route.

Reply:
@brianquinlan I found out it is one not 2 when etag value is inserted.